pkg/odo/cli/dev/dev.go

```diff
@@ -267,6 +281,13 @@ func printPortForwardingInfo(portPairs map[string][]string, out io.Writer) {
 	fmt.Fprintf(out, "\n%s", portForwardURLs.String())
 }
 
+func (o *DevOptions) HandleSignal() error {
+	o.cancel()
+	<-o.cleanupDone
+	time.Sleep(10 * time.Second)
```

Comment:
@kadel @feloy I have removed this sleep in a subsequent commit, but you could add this to test that when cleanup logic is being executed, it doesn't re-trigger the watch logic.